xdem/coreg/biascorr.py

```diff
@@ -1077,7 +1107,10 @@ def _fit_rst_rst(  # type: ignore
         p0 = np.ones(shape=((self._meta["poly_order"] + 1) ** 2))
 
         # Coordinates (we don't need the actual ones, just array coordinates)
-        xx, yy = np.meshgrid(np.arange(0, ref_elev.shape[1]), np.arange(0, ref_elev.shape[0]))
+        if type(ref_elev) == da.Array:
+            xx, yy = da.meshgrid(da.arange(0, ref_elev.shape[1]), da.arange(0, ref_elev.shape[0]))
```

Comment:
We should define the `chunks` here to ensure the auto ones are not completely irrelevant and result in a lot of memory usage. Maybe simply copying the chunks of the reference DEM?

Reply:
So it seems like chunks is not a supported argument to `da.meshgrid` unfortunately. i tried rechunking it but it doesn't seem to make that big of a difference computation-wise :thinking:

Reply:
Strange indeed, it seems important... We can check if that matters later with a memory usage-specific test!

Reply:
yes i agree! I'll add a note so we remember to have a closer look at it.

Reply:
Amazing!

I still found it weird that they didn't ask a chunk argument directly in Dask, and found this interesting discussion:
https://github.com/dask/dask/pull/2943#issuecomment-350478499.

If I understood correctly, doing `da.meshgrid(da.arange(, chunks=), da.arange(, chunks=)` should have the same behaviour as chunking! (or even more optimized)

Reply:
yes! Adding the chunks to the input `da.arange()` makes it possible to chunk the output array. However, I didn't find a way to combine it with delayed. Meaning even if it's chunked I think it loads the entire array into memory. Instead of "creating"? / "reading" ? that part only when needed in the processing. But I'd love to be proven wrong :smiley:

Reply:
I don't know why exactly but when I use this version of meshgrid the entire application just hangs without doing anything... maybe i'll investigate further but currently I can't seem to figure out what's the issue :thinking: